### PR TITLE
Added debug to i2c_device.

### PR DESCRIPTION
**Pull request by kattni**

### CODE REVIEW

File: adafruit_bus_device/i2c_device.py [hunk 1]
```diff
@@ -147,10 +152,19 @@ def write_then_readinto(self, out_buffer, in_buffer, *,
             self.i2c.writeto_then_readfrom(self.device_address, out_buffer, in_buffer,
                                            out_start=out_start, out_end=out_end,
                                            in_start=in_start, in_end=in_end, stop=stop)
+            if self._debug:
+                print("i2c_device.writeto_then_readfrom.out_buffer:", [hex(i) for i in out_buffer])
+                print("i2c_device.writeto_then_readfrom.in_buffer:", [hex(i) for i in in_buffer])
```
Comment (ladyada) [inline adafruit_bus_device/i2c_device.py]: you need to slice in_start and in_end here, ditto above for out_buffer

**Comment by kattni** [comment]
Tested on Feather M4 with VCNL4040.
```Auto-reload is on. Simply save files over USB to run them or enter REPL to disable.
code.py output:
i2c_device.write: ['0xc', '0x0', '0x0']
i2c_device.readinto: ['0xc', '0x86', '0x1']
i2c_device.write: ['0x3', '0x0', '0x0']
i2c_device.readinto: ['0x3', '0x1', '0x0']
i2c_device.write: ['0x3', '0x1', '0x0']
i2c_device.write: ['0x0', '0x0', '0x1']
```

File: adafruit_bus_device/i2c_device.py [hunk 2]
```diff
@@ -147,10 +152,19 @@ def write_then_readinto(self, out_buffer, in_buffer, *,
...
         else:
             # If we don't have a special implementation, we can fake it with two calls
             self.write(out_buffer, start=out_start, end=out_end, stop=stop)
+            if self._debug:
+                print("i2c_device.write_then_readinto.write.out_buffer:",
+                      [hex(i) for i in out_buffer])
```
Comment (ladyada) [inline adafruit_bus_device/i2c_device.py]: slice here too

File: adafruit_bus_device/i2c_device.py [hunk 3]
```diff
@@ -147,10 +152,19 @@ def write_then_readinto(self, out_buffer, in_buffer, *,
...
             self.readinto(in_buffer, start=in_start, end=in_end)
+            if self._debug:
+                print("i2c_device.write_then_readinto.readinto.in_buffer:",
+                      [hex(i) for i in in_buffer])
```
Comment (ladyada) [inline adafruit_bus_device/i2c_device.py]: n here!

**Review by ladyada — CHANGES_REQUESTED** [review]

File: adafruit_bus_device/i2c_device.py [hunk 4]
```diff
@@ -153,18 +153,20 @@ def write_then_readinto(self, out_buffer, in_buffer, *,
                                            out_start=out_start, out_end=out_end,
                                            in_start=in_start, in_end=in_end, stop=stop)
             if self._debug:
-                print("i2c_device.writeto_then_readfrom.out_buffer:", [hex(i) for i in out_buffer])
-                print("i2c_device.writeto_then_readfrom.in_buffer:", [hex(i) for i in in_buffer])
+                print("i2c_device.writeto_then_readfrom.out_buffer:",
+                      [hex(i) for i in out_buffer[out_start:out_end]])
+                print("i2c_device.writeto_then_readfrom.in_buffer:",
```
Comment (ladyada) [inline adafruit_bus_device/i2c_device.py:159]: i think print the writebuffer slice *before* the command, then the readbuffer after.

**Review by ladyada — CHANGES_REQUESTED** [review]

**Review by tannewt — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**